### PR TITLE
Update publish.yml to build stable version

### PR DESCRIPTION
**Pull request by markusweigelt**

### CODE REVIEW

**Comment by markusweigelt** [comment]
After we have created the stable branch we still need to adjust the docker-compose.yml and add the version argument for the parent version.

**Review by bertsky — APPROVED** [review]
LGTM

**Comment by bertsky** [comment]
> After we have created the stable branch we still need to adjust the docker-compose.yml and add the version argument for the parent version.

You mean the `OCRD_ALL_VERSION`? 

Actually, I now believe we should track that in the repo itself. So the build-args would read from a repo-managed file – say `base.version` – which naturally on master would simply be `maximum-git` but on stable represent a date release which we could edit every once in a while.

**Comment by bertsky** [comment]
> Actually, I now believe we should track that in the repo itself. So the build-args would read from a repo-managed file – say `base.version` – which naturally on master would simply be `maximum-git` but on stable represent a date release which we could edit every once in a while.

Or better yet: track ocrd_all as a subrepo (without checkout or even recursive checkout of course). The same could be done for core in ocrd_manager.

**Comment by markusweigelt** [comment]
This all sounds like a lot of overhead than adding single build arg to docker-compose only in the stable branch.

Btw cool there are ocrd/core version distinctions on Docker Hub now 😎

**Comment by bertsky** [comment]
> This all sounds like a lot of overhead than adding single build arg to docker-compose only in the stable branch.

indeed, forgot about that. Let's use only that then!

Currently, we pass VCS_REF and BUILD_DATE dynamically, and your new `OCRD_ALL_VERSION` gets set on the CI statically. IMO the latter would need to be changed. Instead, set `VERSION` in docker-compose.yml directly, so it can be version controlled (and also applies for local builds via Dockerfile btw).

**Comment by markusweigelt** [comment]
Solved it a bit differently but almost as good (since the current `docker/build-push-action` works so well incl. caching etc.). I hope it fits like this.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -20,13 +20,21 @@ jobs:
...
     steps:
+    - uses: actions/checkout@v3
+
+    - name: Run read-yaml action
+      if: endsWith(github.ref, '/stable')
+      id: docker-compose-data
+      uses: jbutcher5/read-yaml@1.6
+      with:
+        file: './docker-compose.yml'
+        key-path: '["services", "ocrd-controller", "args", "VERSION"]'
```
Comment (bertsky) [inline .github/workflows/publish.yml:31]: nice pattern, but I don't see the need for this. You query the build args in the compose file, write the value into the env file, finally use that variable as **override** value for the build args. But that specific build arg is already (going to be) there, no need to do this extra on the CI.

Comment (markusweigelt) [reply .github/workflows/publish.yml:31]: Currently we are not building with Docker Compose on CI but with a Docker GitHub action that supports caching as well as pushing.

Comment (bertsky) [reply .github/workflows/publish.yml:31]: Oh, right! Sry, got distracted. Looks good!

**Pull request merged**